packages/valory/skills/apy_estimation_abci/io_/store.py

```diff
@@ -0,0 +1,123 @@
+from typing import Any, Callable, Dict, Optional, TypeVar, Union, cast
+
+import joblib
+import pandas as pd
+from pmdarima.pipeline import Pipeline
+
+from packages.valory.skills.abstract_round_abci.io_.store import (
+    AbstractStorer,
+    StoredJSONType,
+)
+from packages.valory.skills.abstract_round_abci.io_.store import Storer as BaseStorer
+
+
+NativelySupportedSingleObjectType = Union[StoredJSONType, Pipeline, pd.DataFrame]
+NativelySupportedMultipleObjectsType = Dict[str, NativelySupportedSingleObjectType]
+NativelySupportedObjectType = Union[
+    NativelySupportedSingleObjectType, NativelySupportedMultipleObjectsType
+]
+NativelySupportedStorerType = Callable[[str, NativelySupportedObjectType, Any], None]
+CustomObjectType = TypeVar("CustomObjectType")
+CustomStorerType = Callable[[str, CustomObjectType, Any], None]
+SupportedSingleObjectType = Union[NativelySupportedObjectType, CustomObjectType]
+SupportedMultipleObjectsType = Dict[str, SupportedSingleObjectType]
+SupportedObjectType = Union[SupportedSingleObjectType, SupportedMultipleObjectsType]
+SupportedStorerType = Union[NativelySupportedStorerType, CustomStorerType]
+NativelySupportedJSONStorerType = Callable[
+    [str, Union[StoredJSONType, Dict[str, StoredJSONType]], Any], None
+]
+NativelySupportedPipelineStorerType = Callable[
+    [str, Union[Pipeline, Dict[str, Pipeline]], Any], None
+]
+NativelySupportedDfStorerType = Callable[
+    [str, Union[pd.DataFrame, Dict[str, pd.DataFrame]], Any], None
+]
+
+
+class ExtendedSupportedFiletype(Enum):
+    """Enum for the supported filetypes of the IPFS interacting methods."""
+
+    PM_PIPELINE = auto()
+    CSV = auto()
+
+
+class CSVStorer(AbstractStorer):
+    """A CSV file storer."""
+
+    def store_single_file(
+        self, filename: str, obj: NativelySupportedSingleObjectType, **kwargs: Any
+    ) -> None:
+        """Store a pandas dataframe."""
+        if not isinstance(obj, pd.DataFrame):
+            raise ValueError(  # pragma: no cover
+                f"`JSONStorer` cannot be used with a {type(obj)}! Only with a {pd.DataFrame}"
+            )
+
+        index = kwargs.get("index", False)
+
+        try:
+            obj.to_csv(filename, index=index)
+        except (TypeError, OSError) as e:  # pragma: no cover
+            raise IOError(str(e)) from e
+
+
+class ForecasterStorer(AbstractStorer):
+    """A pmdarima Pipeline storer."""
+
+    def store_single_file(
+        self, filename: str, obj: NativelySupportedSingleObjectType, **kwargs: Any
+    ) -> None:
+        """Store a pmdarima Pipeline."""
+        if not isinstance(obj, Pipeline):
+            raise ValueError(  # pragma: no cover
+                f"`JSONStorer` cannot be used with a {type(obj)}! Only with a {Pipeline}"
+            )
+
+        try:
+            joblib.dump(obj, filename)
+        except (ValueError, OSError) as e:  # pragma: no cover
+            raise IOError(str(e)) from e
+
+
+class Storer(BaseStorer):
```

Comment:
another `dict`

Reply:
Why do you think it is a single `dict`? It has `store_single_file` and `store`.